unitary/examples/quantum_chinese_chess/move.py

```diff
@@ -176,3 +176,57 @@ def effect(self, *objects) -> Iterator[cirq.Operation]:
         target_0.reset(source_0)
         source_0.reset()
         return iter(())
+
+
+class SplitJump(QuantumEffect):
+    """SplitJump from source_0 to target_0 and target_1. The only accepted (default) move_variant is
+    - BASIC.
+    """
+
+    def __init__(
+        self,
+    ):
+        return
+
+    def num_dimension(self) -> Optional[int]:
+        return 2
+
+    def num_objects(self) -> Optional[int]:
+        return 3
+
+    def effect(self, *objects) -> Iterator[cirq.Operation]:
+        source_0, target_0, target_1 = objects
+        # Make the split jump.
+        source_0.is_entangled = True
+        alpha.PhasedSplit()(source_0, target_0, target_1)
+        # Pass the classical properties of the source piece to the target pieces.
+        target_0.reset(source_0)
+        target_1.reset(source_0)
+        return iter(())
+
+
+class MergeJump(QuantumEffect):
+    """MergeJump from source_0 to source_1 to target_0. The only accepted (default) move_variant is
+    - BASIC.
+    """
+
+    def __init__(
+        self,
+    ):
+        return
+
+    def num_dimension(self) -> Optional[int]:
+        return 2
+
+    def num_objects(self) -> Optional[int]:
+        return 3
+
+    def effect(self, *objects) -> Iterator[cirq.Operation]:
+        source_0, source_1, target_0 = objects
+        # Make the merge jump.
+        alpha.PhasedMove(-0.5)(source_0, target_0)
+        alpha.PhasedMove(-0.5)(source_0, target_0)
+        alpha.PhasedMove(-0.5)(source_1, target_0)
+        # Pass the classical properties of the source pieces to the target piece.
+        target_0.reset(source_0)
```

Comment:
This semantic is kind of confusing.  I am not totally sure if this is correct.  Remind me what reset is supposed to do again?

Reply:
It copies the "type", "color" and "is_entangled" properties from source_0 to target_0.

Reply:
I see.  And we are assuming that type, color are the same for source_0 and source_1.

Couldn't is_entangled be different for source_0 and source_1 though?
Also, do we ever fix the type/color for the source squares?  Like, they may be empty now, right?

I don't think this logic is correct.

Reply:
Hi Doug, yes, earlier in chess.py we've checked to make sure source_0 and source_1 have the same type, color, and is_entangled = true (maybe later we could support classical to merge with entangled piece).

The type/color/is_entangled for source squares are fixed through the method "update_board_by_sampling()" in chess.py, which will be called after each move. There are cases that they won't be reset to empty (see test cases for those scenarios).

The source piece for SplitJump will also be fixed through update_board_by_sampling().